src/test/java/fr/insee/pearljam/api/campaign/controller/VisibilityControllerTest.java

```diff
@@ -0,0 +1,152 @@
+package fr.insee.pearljam.api.campaign.controller;
+
+import fr.insee.pearljam.api.campaign.controller.dummy.VisibilityFakeService;
+import fr.insee.pearljam.api.campaign.dto.input.VisibilityUpdateDto;
+import fr.insee.pearljam.api.campaign.dto.output.VisibilityCampaignDto;
+import fr.insee.pearljam.api.utils.AuthenticatedUserTestHelper;
+import fr.insee.pearljam.api.utils.JsonTestHelper;
+import fr.insee.pearljam.api.utils.MockMvcTestUtils;
+import fr.insee.pearljam.api.utils.dummy.AuthenticationUserFakeService;
+import fr.insee.pearljam.api.web.exception.ExceptionControllerAdvice;
+import fr.insee.pearljam.domain.campaign.model.Visibility;
+import fr.insee.pearljam.domain.exception.CampaignNotFoundException;
+import fr.insee.pearljam.domain.exception.VisibilityHasInvalidDatesException;
+import fr.insee.pearljam.domain.exception.VisibilityNotFoundException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+class VisibilityControllerTest {
+
+    private MockMvc mockMvc;
+    private VisibilityFakeService visibilityService;
+    private final String campaignId = "campaign-id";
+    private final String organizationalUnitId = "ou-id";
+    private String updatePath;
+    private String findPath;
+
+    @BeforeEach
+    void setup() {
+        updatePath = String.format("/api/campaign/%s/organizational-unit/%s/visibility", campaignId, organizationalUnitId);
+        findPath = String.format("/api/campaign/%s/visibilities", campaignId);
```

Comment:
paths are formatted with final String and is not updated in tests => could be initiated once for all and not in `setup`